### PR TITLE
Add section on smart contract architecture to Engine 101

### PR DESCRIPTION
**Pull request by jakerockland**
Adding a summary of the content in https://github.com/ArtBlocks/artblocks-contracts/blob/main/V3_ARCHITECTURE.md to the docs here.

### CODE REVIEW

File: art-blocks-engine-onboarding/art-blocks-engine-101/what is Art Blocks Engine.md [hunk 1]
```diff
@@ -26,6 +26,25 @@ Note that for a single project, it is possible to have a single off-chain depend
...
 
+## What is the smart contract architecture for Art Blocks Engine?
+
+The Art Blocks Engine offers two core contract options: the V3 Engine core contract and the V3 Engine Flex core contract. These contracts are mutually exclusive, and partners should select the appropriate core contract based on their needs and whether they require the flex capabilities.
+
+The V3 Engine core contract is an ERC-721 NFT contract that manages metadata for all Art Blocks NFTs, including artist scripts, token hashes, and token royalty data.
+
+The V3 Engine Flex core contract allows artists to use external assets in their Engine tokens. These external assets can be images, videos, audio, or other data and may be stored on decentralized storage systems such as IPFS, Arweave, or on the Ethereum blockchain.
```
Comment (ABDruid) [inline art-blocks-engine-onboarding/art-blocks-engine-101/what is Art Blocks Engine.md:35]: The V3 Engine Flex core contract includes everything in the `V3 Engine` contract, but also allows artists to use external assets in their Engine tokens. These external assets can be images, videos, audio, or other data and may be stored on decentralized storage systems such as IPFS, Arweave, or on the Ethereum blockchain

**Review by ABDruid — COMMENTED** [review]
idk how to submit a change as a reviewer, but I'd like to add a clarification to line 35

"The V3 Engine Flex core contract includes everything in the V3 Engine contract, but also allows artists to use external assets in their Engine tokens"

**Comment by jakerockland** [comment]
> idk how to submit a change as a reviewer, but I'd like to add a clarification to line 35
> 
> "The V3 Engine Flex core contract includes everything in the V3 Engine contract, but also allows artists to use external assets in their Engine tokens"

Updated to include your feedback ser!

For future reference, this lil guide helps clarify how you can suggest changes in PRs directly if that is easier workflow in the future: https://egghead.io/lessons/github-add-suggestions-in-a-github-pr-review 💜

File: art-blocks-engine-onboarding/art-blocks-engine-101/what is Art Blocks Engine.md [hunk 2]
```diff
@@ -26,6 +26,44 @@ Note that for a single project, it is possible to have a single off-chain depend
...
+- Engine Registry contract: Notifies the subgraph indexing service of new Art Blocks Engine tokens. When the Engine Registry emits an event, the subgraph indexing service is notified, and the Engine contract is indexed and made available for querying on the Art Blocks subgraph.
+- Minter Suite contracts: A collection of contracts used to mint Art Blocks Engine tokens. The Minter Suite is designed to be highly flexible and can be used to mint tokens in various ways.
+
+Partners should choose between the V3 Engine core contract and the V3 Engine Flex core contract based on their project's goals and technical capabilities. The smart contract architecture for Art Blocks Engine provides a robust and flexible system for managing and creating generative NFTs while integrating with various peripheral contracts to extend its capabilities.
+
+For additional context, please check out [this architecture overview (with accompanying diagrams)](https://github.com/ArtBlocks/artblocks-contracts/blob/main/MINTER_SUITE.md).
+
+## What is the "minter suite"?
```
Comment (jakerockland) [inline art-blocks-engine-onboarding/art-blocks-engine-101/what is Art Blocks Engine.md:48]: @ABDruid I also added another section here specifically clarifying what the minter suite is.

cc @ryley-o

**Review by ABDruid — APPROVED** [review]

**Review by ryley-o — APPROVED** [review]
love it 👏

**Pull request merged**